presto-spi/src/main/java/io/prestosql/spi/block/MapBlockEncoding.java

```diff
@@ -84,7 +75,7 @@ public void writeBlock(BlockEncodingSerde blockEncodingSerde, SliceOutput sliceO
     @Override
     public Block readBlock(BlockEncodingSerde blockEncodingSerde, SliceInput sliceInput)
     {
-        MapType mapType = (MapType) TypeSerde.readType(typeManager, sliceInput);
+        MapType mapType = (MapType) blockEncodingSerde.readType(sliceInput);
```

Comment:
Preexisting, but it seems weird to decode type from the data stream, since type should be statically known.
Do we have plans to change this?